### PR TITLE
feat(4chan): add presence

### PR DESCRIPTION
**Pull request by beardotpy**
## Description 

A presence for 4chan.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

</details>

### CODE REVIEW

File: websites/0-9/4chan/presence.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
+const presence = new Presence({
+	clientId: "1195051461822914640",
+});
+
+const enum Assets {
+	Cog = "https://i.imgur.com/XAgEs5E.png",
```
Comment (DarkVillager) [inline websites/0-9/4chan/presence.ts:6]: It must be named Logo & also must be 512x512 just like the metadata logo
```suggestion
	Logo = "https://i.imgur.com/7pI9X5M.png",
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/0-9/4chan/presence.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		"/wsr/": "Worksafe Requests",
+		"/x/": "Paranormal",
+		"/xs/": "Extreme Sports",
+		"/y/": "Yaoi",
+	},
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Cog,
```
Comment (DarkVillager) [inline websites/0-9/4chan/presence.ts:93]:
```suggestion
		largeImageKey: Assets.Logo,
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/0-9/4chan/presence.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		"/xs/": "Extreme Sports",
+		"/y/": "Yaoi",
+	},
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+		startTimestamp: browsingTimestamp,
+	};
```
Comment (DarkVillager) [inline websites/0-9/4chan/presence.ts:95]:
```suggestion
	},
	{ pathname } = document.location;
```

File: websites/0-9/4chan/presence.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+		if (document.location.pathname.includes("/thread/")) {
+			const threadNum = document.location.pathname.split("/").at(-1),
+				threadSubject = document.querySelector(".subject").textContent;
+
+			if (threadSubject)
+				presenceData.state = `>>${threadNum} - "${threadSubject}"`;
+			else presenceData.state = `>>${threadNum}`;
+		} else if (document.location.pathname.includes("/archive"))
+			presenceData.state = "Checking the archive";
+	}
```
Comment (DarkVillager) [inline websites/0-9/4chan/presence.ts:120]:
```suggestion
	if (pathname === "/")
		presenceData.details = "Viewing the front page";
	else if (pathname.startsWith("/faq"))
		presenceData.details = "Reading the FAQ";
	else if (pathname.startsWith("/rules"))
		presenceData.details = "Reading the rules";
	else if (pathname.startsWith("/4channews"))
		presenceData.details = "Viewing news posts";
	else {
		for (const board in boards) {
			if (pathname.includes(board))
				presenceData.details = `Browsing ${board} - ${boards[board]}`;
		}

		if (pathname.includes("/thread/")) {
			const threadNum = pathname.split("/").at(-1),
				threadSubject = document.querySelector(".subject").textContent;

			if (threadSubject)
				presenceData.state = `>>${threadNum} - "${threadSubject}"`;
			else presenceData.state = `>>${threadNum}`;
		} else if (pathname.includes("/archive"))
			presenceData.state = "Checking the archive";
	}
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/0-9/4chan/metadata.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.9",
+	"author": {
+		"id": "623396579960946690",
+		"name": "bear_"
+	},
+	"service": "4chan",
+	"description": {
+		"en": "4chan is an anonymous English-language imageboard website. Launched by Christopher \\\"moot\\\" Poole in October 2003, the site hosts boards dedicated to a wide variety of topics, from video games and television to literature, cooking, weapons, music, history, anime, fitness, politics, and sports, among others."
+	},
+	"url": "4chan.org",
```
[the remark this answers is not in the document]
Comment (beardotpy) [reply websites/0-9/4chan/metadata.json:11]: fix(4chan): add support for www domain

**Comment by beardotpy** [comment]
it was more than a couple hours...sorry

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**